inc/formatconvert.class.php

```diff
@@ -1935,6 +1927,42 @@ static function networkequipmentInventoryTransformation($array) {
       $array_tmp['last_fusioninventory_update'] = date('Y-m-d H:i:s');
       $a_inventory['PluginFusioninventoryNetworkEquipment'] = $array_tmp;
 
+      //manage firmwares
+      //['INFO']['FIRMWARE'] is the only available value until FI agent 2.3.21
+      //['FIRWARES'] (multivalued) will be available in next releases
+      if (isset($array['INFO']['FIRMWARE']) || isset($array['FIRMWARES'])) {
+         $a_firmwares = [];
+         $mapping = [
+            'DESCRIPTION'  => 'description',
+            'MANUFACTURER' => 'manufacturers_id',
+            'NAME'         => 'name',
+            'TYPE'         => 'devicefirmwaretypes_id',
+            'VERSION'      => 'version'
+         ];
+         if (isset($array['FIRMWARES'])) {
```

Comment:
I would suggest, if FIRMWARES is not an array, to create and array and manage only one case